lua/nvim-treesitter/textobjects/move.lua

```diff
@@ -9,11 +9,13 @@ local M = {}
 local function move(query_strings, forward, start, bufnr)
   query_strings = shared.make_query_strings_table(query_strings)
   bufnr = bufnr or vim.api.nvim_get_current_buf()
-  local row, col = unpack(vim.api.nvim_win_get_cursor(0))
-  row = row - 1 -- nvim_win_get_cursor is (1,0)-indexed
 
+  local config = configs.get_module "textobjects.move"
   local function filter_function(match)
     local range = { match.node:range() }
+    local row, col = unpack(vim.api.nvim_win_get_cursor(0))
```

Comment:
This kind of contradicts the `buf` argument (when always using 0). At the moment we didn't expose `buf` publicly. Can you change the 4th argument of this function to `win` and use `nvim_win_get_buf` to get the buffer for the window `win`. `nvim_get_current_win()` can be used if `win` is nil